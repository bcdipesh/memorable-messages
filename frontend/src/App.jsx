import { BrowserRouter } from "react-router-dom";

import AuthProvider from "@/authProvider/AuthProvider";
import { ThemeProvider } from "@/components/ThemeProvider";
import AppRoutes from "@/components/routing/AppRoutes";
import { Toaster } from "@/components/ui/toaster";

/**
 * The main App component that wraps the entire application.
 * It provides the necessary context providers and sets up routing.
 *
 * @returns {React.JSX.Element} The App component.
 */
const App = () => {
  return (
    // ThemeProvider wraps the entire application to provide system wide dark and light theme
    <ThemeProvider defaultTheme="dark" storageKey="vite-ui-theme">
      <div className="app">
        {/* AuthProvider wraps the application with authentication context */}
        <AuthProvider>
          {/* BrowserRouter provides the navigation context for routing */}
          <BrowserRouter>
            {/* AppRoutes contains the routing configuration for the application */}
            <AppRoutes />
            {/* Toaster displays toast notifications for the application */}
            <Toaster />
          </BrowserRouter>
        </AuthProvider>
      </div>
    </ThemeProvider>
  );
};

export default App;
