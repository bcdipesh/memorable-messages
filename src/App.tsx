import { createBrowserRouter, RouterProvider } from "react-router-dom";

import Root from "./pages/Root";
import HomePage from "./pages/HomePage";
import SignInPage from "./pages/SignInPage";
import SignUpPage from "./pages/SignUpPage";
import TermsOfServicePage from "./pages/TermsOfServicePage";
import PrivacyPolicyPage from "./pages/PrivacyPolicyPage";

const router = createBrowserRouter([
  {
    path: "/",
    element: <Root />,
    children: [
      {
        index: true,
        element: <HomePage />,
      },
      {
        path: "/sign-in/*",
        element: <SignInPage />,
      },
      {
        path: "/sign-up/*",
        element: <SignUpPage />,
      },
      {
        path: "/terms-of-service",
        element: <TermsOfServicePage />,
      },
      {
        path: "/privacy-policy",
        element: <PrivacyPolicyPage />,
      },
    ],
  },
]);

export default function App() {
  return <RouterProvider router={router} />;
}
