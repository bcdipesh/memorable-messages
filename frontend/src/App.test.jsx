import React from "react";

import { render } from "@testing-library/react";

import App from "./App";

test("Renders without crashing", function () {
  render(<App />);
});

test("Matches snapshot", function () {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});
