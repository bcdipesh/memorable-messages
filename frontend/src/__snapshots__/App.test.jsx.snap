// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <div
      class="layout container flex h-screen flex-col justify-between"
    >
      <nav
        class="mt-6 flex list-none flex-col justify-between space-y-6 md:flex-row md:space-y-0"
      >
        <a
          href="/"
        >
          <div
            class="logo flex items-center space-x-3"
          >
            <svg
              class="h-6 w-6"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4.913 2.658c2.075-.27 4.19-.408 6.337-.408 2.147 0 4.262.139 6.337.408 1.922.25 3.291 1.861 3.405 3.727a4.403 4.403 0 0 0-1.032-.211 50.89 50.89 0 0 0-8.42 0c-2.358.196-4.04 2.19-4.04 4.434v4.286a4.47 4.47 0 0 0 2.433 3.984L7.28 21.53A.75.75 0 0 1 6 21v-4.03a48.527 48.527 0 0 1-1.087-.128C2.905 16.58 1.5 14.833 1.5 12.862V6.638c0-1.97 1.405-3.718 3.413-3.979Z"
              />
              <path
                d="M15.75 7.5c-1.376 0-2.739.057-4.086.169C10.124 7.797 9 9.103 9 10.609v4.285c0 1.507 1.128 2.814 2.67 2.94 1.243.102 2.5.157 3.768.165l2.782 2.781a.75.75 0 0 0 1.28-.53v-2.39l.33-.026c1.542-.125 2.67-1.433 2.67-2.94v-4.286c0-1.505-1.125-2.811-2.664-2.94A49.392 49.392 0 0 0 15.75 7.5Z"
              />
            </svg>
            <span
              class="text-xl"
            >
              Memorable Messages
            </span>
          </div>
        </a>
        <div
          class="flex space-x-6"
        >
          <ul
            class="flex space-x-6"
          >
            <li>
              <a
                class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2"
                href="/profile"
              >
                Profile
              </a>
            </li>
            <li>
              <a
                class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2"
                href="/occasions"
              >
                Occasions
              </a>
            </li>
            <li>
              <button
                class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2"
              >
                Logout
              </button>
            </li>
            <li>
              <button
                aria-expanded="false"
                aria-haspopup="menu"
                class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 hover:bg-accent hover:text-accent-foreground h-10 w-10"
                data-state="closed"
                id="radix-:r2:"
                type="button"
              >
                <svg
                  class="lucide lucide-sun h-[1.2rem] w-[1.2rem] rotate-0 scale-100 transition-all dark:-rotate-90 dark:scale-0"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="4"
                  />
                  <path
                    d="M12 2v2"
                  />
                  <path
                    d="M12 20v2"
                  />
                  <path
                    d="m4.93 4.93 1.41 1.41"
                  />
                  <path
                    d="m17.66 17.66 1.41 1.41"
                  />
                  <path
                    d="M2 12h2"
                  />
                  <path
                    d="M20 12h2"
                  />
                  <path
                    d="m6.34 17.66-1.41 1.41"
                  />
                  <path
                    d="m19.07 4.93-1.41 1.41"
                  />
                </svg>
                <svg
                  class="lucide lucide-moon absolute h-[1.2rem] w-[1.2rem] rotate-90 scale-0 transition-all dark:rotate-0 dark:scale-100"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 3a6 6 0 0 0 9 9 9 9 0 1 1-9-9Z"
                  />
                </svg>
                <span
                  class="sr-only"
                >
                  Toggle theme
                </span>
              </button>
            </li>
          </ul>
        </div>
      </nav>
      <div
        class="home mt-24 grid space-y-10 md:mt-0 md:grid-cols-2 md:gap-x-6 md:space-y-0"
      >
        <div
          class="flex flex-col space-y-6"
        >
          <h1
            class="text-3xl font-bold"
          >
            Welcome back, null
          </h1>
          <p>
            So excited to see you again, null! We're here to help you make every celebration unforgettable.
          </p>
        </div>
        <img
          alt="A girl sending wishes"
          class="md:row-span-2"
          src="/src/assets/hero.svg"
        />
        <div
          class="flex flex-col space-y-6 md:row-start-2 md:row-end-2"
        >
          <h2
            class="text-xl"
          >
            Create a new message and bring joy to someone's day!
          </h2>
          <a
            class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2 w-fit"
            href="/occasions"
          >
            Create
          </a>
        </div>
      </div>
      <footer
        class="flex flex-col space-y-6 py-16"
      >
        <div
          class="shrink-0 bg-border h-[1px] w-full"
          data-orientation="horizontal"
          role="none"
        />
        <div
          class="flex flex-col justify-between md:flex-row"
        >
          <a
            href="/"
          >
            <div
              class="logo flex items-center space-x-3"
            >
              <svg
                class="h-6 w-6"
                fill="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4.913 2.658c2.075-.27 4.19-.408 6.337-.408 2.147 0 4.262.139 6.337.408 1.922.25 3.291 1.861 3.405 3.727a4.403 4.403 0 0 0-1.032-.211 50.89 50.89 0 0 0-8.42 0c-2.358.196-4.04 2.19-4.04 4.434v4.286a4.47 4.47 0 0 0 2.433 3.984L7.28 21.53A.75.75 0 0 1 6 21v-4.03a48.527 48.527 0 0 1-1.087-.128C2.905 16.58 1.5 14.833 1.5 12.862V6.638c0-1.97 1.405-3.718 3.413-3.979Z"
                />
                <path
                  d="M15.75 7.5c-1.376 0-2.739.057-4.086.169C10.124 7.797 9 9.103 9 10.609v4.285c0 1.507 1.128 2.814 2.67 2.94 1.243.102 2.5.157 3.768.165l2.782 2.781a.75.75 0 0 0 1.28-.53v-2.39l.33-.026c1.542-.125 2.67-1.433 2.67-2.94v-4.286c0-1.505-1.125-2.811-2.664-2.94A49.392 49.392 0 0 0 15.75 7.5Z"
                />
              </svg>
              <span
                class="text-xl"
              >
                Memorable Messages
              </span>
            </div>
          </a>
          <div
            class="flex space-x-6"
          >
            <a
              class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 text-primary underline-offset-4 hover:underline h-10 w-fit p-0"
              href="/terms-of-service"
            >
              Terms of Service
            </a>
            <a
              class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 text-primary underline-offset-4 hover:underline h-10 w-fit p-0"
              href="/privacy-policy"
            >
              Privacy Policy
            </a>
          </div>
        </div>
        <p
          class="text-sm text-muted-foreground"
        >
          Â© 2024 Memorable Messages. Made to create memories that will last a lifetime.
        </p>
      </footer>
    </div>
    <div
      aria-label="Notifications (F8)"
      role="region"
      style="pointer-events: none;"
      tabindex="-1"
    >
      <ol
        class="fixed top-0 z-[100] flex max-h-screen w-full flex-col-reverse p-4 sm:bottom-0 sm:right-0 sm:top-auto sm:flex-col md:max-w-[420px]"
        tabindex="-1"
      />
    </div>
  </div>
</DocumentFragment>
`;
